src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -206,18 +206,18 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
 
         public void Save(IConnectionViewModel connectionViewModel) {
             _shell.AssertIsOnMainThread();
-            if (connectionViewModel == null) {
+            if (connectionViewModel == null || !connectionViewModel.HasChanges) {
                 return;    
             }
 
-            var connection = _connectionManager.AddOrUpdateConnection(
+            _connectionManager.AddOrUpdateConnection(
```

Comment:
Frankly in UI list with in-place editing would be simpler than the dictionary ;-) It's like 5 elements max.